### DISCUSSION TITLE
Openvpn Connect Cask - Updating removes installed profiles

### DISCUSSION THREAD

**Discussion by ben-ihelputech** [Homebrew/discussions#4087] [Casks]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 3.6.17
ORIGIN: https://github.com/Homebrew/brew
HEAD: e76c55e18434c870cca1de7aadf9d5218271fe2d
Last commit: 7 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 3145eebc9d223953120e4acb2ff29a1e3da27866
Core tap last commit: 5 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: vim
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit skylake
Clang: 14.0.0 build 1400
Git: 2.37.1 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.79.1 => /usr/bin/curl
macOS: 12.6.1-x86_64
CLT: 14.1.0.0.1.1666437224
Xcode: 14.2
```

### Output of `brew doctor`

```shell
﻿$ brew doctor         
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  ansible

Warning: Homebrew's "sbin" was not found in your PATH but you have installed
formulae that put executables in /usr/local/sbin.
Consider setting your PATH for example like so:
  echo 'export PATH="/usr/local/sbin:$PATH"' >> ~/.zshrc

Warning: Broken symlinks were found. Remove them with `brew cleanup`:
  /usr/local/opt/kubectl@1.23
```

## Upgrading OpenVPN Connect Deletes Installed Profiles

Whenever I update Openvpn connect cask (`brew upgrade openvpn-connect`), it does not seem to make a backup of the installed profiles. It seems that the only update path it uses is to completely uninstall the app then reinstall the latest version. This is rather annoying because I have to go and manually reinstall each profile. Before I opened an issue, I wanted to see if anyone else experiences this problem and if it is an issue with the way the cask it written or a problem with the openvpn-connect app itself.

It seems like the issue was raised once before on [#103721](https://github.com/Homebrew/homebrew-cask/issues/103721) but never resolved.

### Relevant casks

openvpn-connect

**Comment by p-linnane** [comment]
Where does OpenVPN store profiles?

**Reply by ben-ihelputech** [reply]
It seems like it stores it in: ~/Library/Application Support/OpenVPN Connect/profiles However, there may be some other items in the Application Support folder that would be worth savings like config.json and settings.json. I guess would there be any reason to not keep the contents of the Application Support folder?

**Reply by p-linnane** [reply]
I've tested this locally and can't reproduce the issue. `"~/Library/Application Support/OpenVPN Connect"` is referenced by the `zap` stanza, but that is only called when passing `--zap` during uninstall. What command are you running?

**Reply by ben-ihelputech** [reply]
Just `brew upgrade openvpn-connect` or `brew upgrade`
